compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -4546,14 +4546,11 @@ private void validateReferencedFunction(Location pos, BAttachedFunction func, Sy
             return;
         }
 
-        // Service typing does not consider resource methods when type checking.
-        if (Symbols.isResource(func.symbol)) {
-            return;
-        }
-
         // There must be an implementation at the outer level, if the function is an interface.
         if (!env.enclPkg.objAttachedFunctions.contains(func.symbol)) {
-            dlog.error(pos, code, func.funcName, func.symbol.receiverSymbol.type);
+            // Here we use the whole function in the error msg since resource function name will contain `$`s if 
+            // we use the func.funcName
```

Comment:
Shall we do this only for resource methods? It complicates/lengthens error messages for normal methods (the common case) and remote methods.

Reply:
But that will give inconsistent error messages for resource functions and other functions?

Reply:
That is better than complicating error messages for the common case IMO.

Reply:
And they are different from other methods anyway. Even for resource methods, we can see if we can improve the error to not include stuff that doesn't actually help to identify the method specifically (like params), and make them more uniform with other methods.